### ISSUE TITLE
test_enable_message_block fails for try/except pragma scope in 4.0.5

### ISSUE THREAD

**Issue by mcepl** [PyCQA/pylint#11009]
### Bug description

```python
`tests/lint/unittest_lint.py::test_enable_message_block` fails in `pylint 4.0.5`:

assert linter.is_message_enabled("E1101", 64)
AssertionError: assert False
The failing case is meth6 in tests/regrtest_data/func_block_disable_msg.py:
def meth6(self):
    # pylint: disable=E1101
    print(self.bla)
    try:
        # pylint: enable=E1101
        print(self.blip)   # expected enabled
    except UndefinedName:  # pylint: disable=E0602
        print(self.blip)   # test expects enabled here too
    print(self.blip)       # expected disabled again
The test expects:
assert not linter.is_message_enabled("E1101", 57)
assert linter.is_message_enabled("E1101", 61)
assert linter.is_message_enabled("E1101", 64)
assert not linter.is_message_enabled("E1101", 66)
but line 64 is reported as disabled.
```

### Configuration

```ini

```

### Command used

```shell
RUnning the test suite
```

### Pylint output

```python
E1101 is enabled at line 61, but still disabled at line 64.
Failure excerpt
___________________________________ FAILURES ___________________________________
___________________________ test_enable_message_block ___________________________
    # meth6
    assert not linter.is_message_enabled("E1101", 57)
    assert linter.is_message_enabled("E1101", 61)
>   assert linter.is_message_enabled("E1101", 64)
E   AssertionError: assert False
tests/lint/unittest_lint.py:320: AssertionError
```

### Expected behavior

```
# pylint: enable=E1101
```

inside the try block should make E1101 enabled in the corresponding except body as well, matching the current test expectation.

### Pylint version

```shell
Observed while packaging for openSUSE OBS with:
- pylint 4.0.5
- astroid 4.2.0
- failure seen on Python 3.11 ([gw2] linux -- Python 3.11.15 /usr/bin/python3.11)
The buildroot used:
- python311-astroid-4.2.0-136.1
- python313-astroid-4.2.0-136.1
- python314-astroid-4.2.0-136.1
```

### OS / Environment

openSUSE/Factory while building the packages for openSUSE

### Additional dependencies

```python

```

**Comment by DanielNoord** [comment]
As can be seen in the [dependencies](https://github.com/pylint-dev/pylint/blob/v4.0.5/pyproject.toml) declaration of the `4.0.5` tag you are using a version of `astroid` that is not supported by that `pylint` version.
